### PR TITLE
Bump dependency versions and fix Android Studio warnings

### PR DESCRIPTION
**Pull request by FrazerSeymour**
What it says on the tin - this MR updates the Gradle version, RightMesh Library and Plugin versions, and specifies a support library version so that Android Studio doesn't complain on compile.

It also updates the language version to 1.8, and uses lambdas and method references everywhere it can.

Finally, I went through and fixed everything Android Studio gave a warning/suggestion about.

### CODE REVIEW

**Review by raycho — APPROVED** [review]

File: app/build.gradle [hunk 1]
```diff
@@ -19,7 +19,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'io.left.rightmesh:rightmesh-plugin:1.3'
+        classpath 'io.left.rightmesh:rightmesh-plugin:1.4'
```
Comment (fractalic) [inline app/build.gradle]: plugin 1.6 now

File: app/build.gradle [hunk 2]
```diff
@@ -58,12 +58,17 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+    compileOptions {
+        targetCompatibility 1.8
+        sourceCompatibility 1.8
```
Comment (fractalic) [inline app/build.gradle:63]: does this still work with old versions of android, or does language version not matter?

**Review by fractalic — CHANGES_REQUESTED** [review]
Some minor q's

File: app/build.gradle (hunk 2 above)
Comment (FrazerSeymour) [reply app/build.gradle:63]: Yep, still works. Or at least works on the `4.4.something` Motorolla phones we have kicking around.

Comment (FrazerSeymour) [reply app/build.gradle:63]: We have a minimum supported SDK thing set in our library and it doesn't complain about it.

File: app/build.gradle (hunk 1 above)
Comment (FrazerSeymour) [reply app/build.gradle]: That was quick.

**Review by ConnorHeidema — APPROVED** [review]

**Review by fractalic — APPROVED** [review]

**Pull request merged**